OpenRA.Platforms.Default/Sdl2PlatformWindow.cs

```diff
@@ -510,18 +517,21 @@ static void SetSDLAttributes(GLProfile profile)
 			}
 		}
 
-		static bool CanCreateGLWindow(GLProfile profile)
+		static bool CanCreateGLWindow(GLProfile profile, List<string> errorLog)
 		{
 			// Implementation inspired by TestIndividualGLVersion from Veldrid
 
 			// Need to create and destroy its own SDL contexts as a workaround for specific buggy drivers
-			SDL.SDL_Init(SDL.SDL_INIT_VIDEO);
+			if (SDL.SDL_Init(SDL.SDL_INIT_VIDEO) != 0)
+				errorLog.Add($"{profile}: SDL init failed: {SDL.SDL_GetError()}");
+
 			SetSDLAttributes(profile);
 
 			var flags = SDL.SDL_WindowFlags.SDL_WINDOW_HIDDEN | SDL.SDL_WindowFlags.SDL_WINDOW_OPENGL;
 			var window = SDL.SDL_CreateWindow("", 0, 0, 1, 1, flags);
 			if (window == IntPtr.Zero || !string.IsNullOrEmpty(SDL.SDL_GetError()))
 			{
+				errorLog.Add($"{profile}: {SDL.SDL_GetError()}");
 				SDL.SDL_ClearError();
 				SDL.SDL_Quit();
 				return false;
```

Comment:
We should also log the result of `SDL_GetError` if `SDL_GL_MakeCurrent` fails a few lines below.